pkg/atconf/file_writer.go

```diff
@@ -0,0 +1,339 @@
+/*
+ (c) Copyright [2021] Micro Focus or one of its affiliates.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ You may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package atconf
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+
+	configparser "github.com/bigkevmcd/go-configparser"
+	"github.com/go-logr/logr"
+	"github.com/lithammer/dedent"
+	vapi "github.com/vertica/vertica-kubernetes/api/v1beta1"
+	"github.com/vertica/vertica-kubernetes/pkg/cmds"
+	"github.com/vertica/vertica-kubernetes/pkg/names"
+	"github.com/vertica/vertica-kubernetes/pkg/net"
+	"github.com/vertica/vertica-kubernetes/pkg/paths"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	ClusterSection          = "Cluster"
+	NodesSection            = "Nodes"
+	ConfigurationSection    = "Configuration"
+	ClusterHostOption       = "hosts"
+	ConfigurationIPv6Option = "ipv6"
+)
+
+// FileWriter is a writer for admintools.conf
+type FileWriter struct {
+	Log            logr.Logger
+	PRunner        cmds.PodRunner
+	Vdb            *vapi.VerticaDB
+	ATConfTempFile string
+	Cfg            *configparser.ConfigParser
+}
+
+// MakeFileWriter will build and return the FileWriter struct
+func MakeFileWriter(log logr.Logger, vdb *vapi.VerticaDB, prunner cmds.PodRunner) Writer {
+	return &FileWriter{
+		Log:     log,
+		Vdb:     vdb,
+		PRunner: prunner,
+	}
+}
+
+// AddHosts will had ips to an admintools.conf.  New admintools.conf, stored in
```

Comment:
minor typo: had --> add. 

I'm trying to get more details: is it adding IPs to the hosts = entry in [Cluster] section?

Reply:
Also, does it check duplicate IPs?

Reply:
Edit: nvm, upon looking at the called functions I think my questions are answered.

Reply:
That's right.  We handle duplicate IPs by treating it as a no-op.  I will update the comment here.